pkg/provider/kubernetes/ingress/client.go

```diff
@@ -307,6 +330,16 @@ func (c *clientWrapper) GetSecret(namespace, name string) (*corev1.Secret, bool,
 	return secret, exist, err
 }
 
+func (c *clientWrapper) GetIngressClass(name string) (*networkingv1beta1.IngressClass, bool, error) {
```

Comment:
alternatively, we could look for `IngressClass` ressources with the controller `traefik.io/ingress-controller` in the spec, no?